driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,322 @@
+
+        double errorsThreshold = 1.0;
+        int retryPeriod = 5000;
+
+        ErrorsFilter errorsFilter = new DefaultErrorsFilter();
+
+        /**
+         * Creates a {@link Builder} instance.
+         *
+         * @param childPolicy the load balancing policy to wrap with error
+         *                    awareness.
+         */
+        public Builder(LoadBalancingPolicy childPolicy) {
+            this.childPolicy = childPolicy;
+        }
+
+        /**
+         * Defines the limit of errors after which a host will be considered INVALID
+         * by the {@link ErrorAwarePolicy}.
+         * <p/>
+         * The error tracker is going to get notified right away, by the {@link Cluster} it is registered to, of each error
+         * happening for each hosts.
+         * <p/>
+         * Internally a {@link Meter} is kept up to date for each host upon each error. The Error aware policy then,
+         * uses the moving average calculated by {@link Meter#getOneMinuteRate} and compares it to the threshold rate
+         * given here in parameter.
+         * <p/>
+         * If the {@link Meter}'s One Minute Rate is <b>higher than or equal to</b> the threshold, the host
+         * will be considered ignored. Otherwise, the host will still be considered valid.
+         * <p/>
+         * Default value for the threshold is 1.
+         *
+         * @param errorsPerSecThreshold the inclusion threshold.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withInclusionThreshold(double errorsPerSecThreshold) {
+            this.errorsThreshold = errorsPerSecThreshold;
+            return this;
+        }
+
+        /**
+         * Provide a class that will allow filtering errors returned by hosts.
+         * <p/>
+         * The {@link ErrorsFilter#shouldConsiderError} method will be called on every error notification the
+         * error tracker associated to the Error aware policy receives. It will then call the error filter, which
+         * will determine whether this error needs to be considered or not.
+         * <p/>
+         * The default implementation will exclude from the error counting, the following exception types :
+         * <ul>
+         *     <li>{@link ReadTimeoutException}</li>
+         *     <li>{@link WriteTimeoutException}</li>
+         *     <li>{@link UnavailableException}</li>
+         *     <li>{@link ReadFailureException}</li>
+         *     <li>{@link WriteFailureException}</li>
+         * </ul>
+         * @param errorsFilter the filter class that the policy will use.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withErrorsFilter(ErrorsFilter errorsFilter) {
+            this.errorsFilter = errorsFilter;
+            return this;
+        }
+
+        /**
+         * Defines the time during which a host is excluded by the tracker.
```

Comment:
Should make it clear that this is in milliseconds, or maybe add a second parameter for TimeUnit so the user can provide the units.